inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/convolution_gpu_b_fs_zyx_fsv16_imad.cl

```diff
@@ -92,8 +113,64 @@ KERNEL(convolution_gpu_b_fs_zyx_fsv16_imad)(
 
     uint4 input_val[IN_BLOCK_DEPTH][IN_BLOCK_HEIGHT][CEIL_DIV(IN_BLOCK_WIDTH, SIMD)];
 
+#if ASYMMETRIC_DATA_QUANTIZATION
+    uint data_zp_idx = g * FILTER_IFM_NUM + in_f_start;
+    uint4 data_zp_val;
+#endif
+#if ASYMMETRIC_WEIGHTS_QUANTIZATION
+    uint4 weights_zp_val[OFM_BLOCKS_PER_SIMD];
+    __attribute__((opencl_unroll_hint))
+    for (uint ofb = 0; ofb < OFM_BLOCKS_PER_SIMD; ++ofb) {
+        weights_zp_val[ofb] = as_uint4((FILTER_TYPE_16)weights_zp[out_f + ofb * FSV]);
```

Comment:
Have you checked how big is overhead for loading per-channel zero-points? Right now we always create per-channel parameters, but actually zero point can be per-tensor, which might be much better from the performance perspective.

Reply:
I haven't run such measurements, I assumes that we need to have per-channel parameters.